### PR TITLE
check on mnemonic when confirming seed

### PR DESCRIPTION
**Pull request by AlaaElattar**
- https://github.com/threefoldtech/threefold_connect/issues/549
- Converting list of entered words to set and check if their length still 3 or not.

### Tested Scenarios:
- tried to enter same word three time.
- tried to entred one word 2 times with another one different.

### CODE REVIEW

**Comment by zaelgohary** [comment]
Please add tested scenarios.

**Review by zaelgohary — APPROVED** [review]

**Comment by AlaaElattar** [comment]
> Please add tested scenarios.

done

File: app/lib/screens/mobile_registration_screen.dart [hunk 1]
```diff
@@ -147,8 +147,16 @@ class _MobileRegistrationScreenState extends State<MobileRegistrationScreen> {
       seedConfirmationController.text = seedCheckValue;
     });
 
+    List<String> enteredWords = seedCheckValue.split(' ');
+    if(enteredWords.length != 3 || enteredWords.toSet().length != 3){
```
Comment (AhmedHanafy725) [inline app/lib/screens/mobile_registration_screen.dart:151]: more than 3 words should be allowed too
```suggestion
    if(enteredWords.length < 3 || enteredWords.toSet().length < 3){
```

File: app/lib/screens/mobile_registration_screen.dart [hunk 2]
```diff
@@ -147,8 +147,16 @@ class _MobileRegistrationScreenState extends State<MobileRegistrationScreen> {
...
+      setState((){
+        errorStepperText = 'Please enter three different words.';
+      });
+      return;
+    }
```
Comment (AhmedHanafy725) [inline app/lib/screens/mobile_registration_screen.dart:156]: let's move this logic inside the `validateSeedWords` method

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

File: app/lib/services/tools_service.dart [hunk 3]
```diff
@@ -30,6 +30,9 @@ bool validateSeedWords(String seed, String confirmationWords) {
 
   // if length is not correct return already here
   if (words.length != 3) return false;
```
Comment (AhmedHanafy725) [inline app/lib/services/tools_service.dart]:
```suggestion
  if (words.length < 3) return false;
```

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**